resolver/resolver.cc

```diff
@@ -3935,6 +3935,82 @@ class ResolveSanityCheckWalk {
     }
 };
 
+class CheckAbstractClassInitialization {
+public:
+    struct CheckAbstractClassInitializationWalkResult {
+        vector<ast::ParsedFile> trees;
+    };
+
+    ast::ExpressionPtr postTransformSend(core::Context ctx, ast::ExpressionPtr tree) {
+        auto &send = ast::cast_tree_nonnull<ast::Send>(tree);
+        if (send.fun != core::Names::new_()) {
+            return tree;
+        }
+
+        auto *id = ast::cast_tree<ast::ConstantLit>(send.recv);
+        if (id == nullptr || !id->symbol.exists() || !id->symbol.isClassOrModule()) {
+            return tree;
+        }
+
+        auto symbol = id->symbol.asClassOrModuleRef().data(ctx);
+        auto singletonClass = symbol->lookupSingletonClass(ctx.state);
+
+        if (!singletonClass.exists()) {
+            return tree;
+        }
+
+        if (symbol->flags.isAbstract) {
+            auto method_new = singletonClass.data(ctx)->findMethodTransitive(ctx.state, core::Names::new_());
+            // If the .new method we find is owned by Class, that means
+            // there was no user defined .new method, which warrants an error.
+            if (method_new.data(ctx)->owner.show(ctx) == "Class") {
```

Comment:
Is there a constant string we can use instead of hardcoding "Class"?

Reply:
Not sure if this is the right answer (still setting things up locally) but would `core::Symbols::Class()` work here?

Reply:
Ah yes, thank you! I also needed to also make it a string - `core::Symbols::Class().toString(ctx)`